modelling/src/main/java/org/axonframework/modelling/command/AnnotationCommandTargetResolver.java

```diff
@@ -70,44 +89,101 @@ public VersionedAggregateIdentifier resolveTarget(CommandMessage<?> command) {
+	}
 
-    private Long findVersion(CommandMessage<?> command) throws InvocationTargetException, IllegalAccessException {
-        for (Method m : methodsOf(command.getPayloadType())) {
-            if (AnnotationUtils.isAnnotationPresent(m, TargetAggregateVersion.class)) {
-                ensureAccessible(m);
-                return asLong(m.invoke(command.getPayload()));
-            }
-        }
-        for (Field f : fieldsOf(command.getPayloadType())) {
-            if (AnnotationUtils.isAnnotationPresent(f, TargetAggregateVersion.class)) {
-                return asLong(getFieldValue(f, command.getPayload()));
-            }
-        }
-        return null;
-    }
+	private Long findVersion(Message<?> command) throws InvocationTargetException, IllegalAccessException {
+		return asLong(invokeAnnotated(command, versionAnnotation));
+	}
 
-    private Long asLong(Object fieldValue) {
-        if (fieldValue == null) {
-            return null;
-        } else if (Number.class.isInstance(fieldValue)) {
-            return ((Number) fieldValue).longValue();
-        } else {
-            return Long.parseLong(fieldValue.toString());
-        }
-    }
-}
+	private static Object invokeAnnotated(Message<?> command, Class<? extends Annotation> annotation)
+			throws InvocationTargetException, IllegalAccessException {
+		for (Method m : methodsOf(command.getPayloadType())) {
+			if (AnnotationUtils.isAnnotationPresent(m, annotation)) {
+				ensureAccessible(m);
+				return m.invoke(command.getPayload());
+			}
+		}
+		for (Field f : fieldsOf(command.getPayloadType())) {
+			if (AnnotationUtils.isAnnotationPresent(f, annotation)) {
+				return getFieldValue(f, command.getPayload());
+			}
+		}
+		return null;
+	}
+
+	private Long asLong(Object fieldValue) {
+		if (fieldValue == null) {
+			return null;
+		} else if (Number.class.isInstance(fieldValue)) {
+			return ((Number) fieldValue).longValue();
+		} else {
+			return Long.parseLong(fieldValue.toString());
+		}
+	}
+
+	@Override
+	public String toString() {
+		return "AnnotationCommandTargetResolver [identifierAnnotation=" + identifierAnnotation + ", versionAnnotation="
+				+ versionAnnotation + "]";
+	}
+
+	public static final class Builder {
+		private AnnotationCommandTargetResolver resolver;
+
+		public Builder() {
+			this.resolver = new AnnotationCommandTargetResolver();
+		}
+
+		/**
+		 * Sets the annotation, that marks the target aggregate identifier.
+		 * <p>
+		 * Defaults to {@link TargetAggregateIdentifier}.<br>
+		 * <p>
+		 * If you do not want your messages-module (as inner bounded context "API") to
+		 * be dependent of axon annotations (e.g. to aim empty pom dependencies), then
+		 * you can write your own annotation based on the
+		 * {@link TargetAggregateIdentifier} without referencing the original one (as
+		 * meta-annotation).
+		 * 
+		 * @param annotation - {@link Class} of type {@link Annotation}.
```

Comment:
Nit: Would you mind removing the `-` from this line here? That way it'll resemble how we typically write the `@param` javadoc lines in the framework.